lib/simple_config.rb

```diff
@@ -75,7 +75,10 @@ def get(key)
     # @return The current value for +:key+.
     def unset(key)
       singleton_class.send(:remove_method, key)
-      @settings.delete(key)
+      setting = @settings.delete(key)
+      # If there was a setting return that, otherwise continue to try and remove a group which may exist
+      return setting if setting.nil?
```

Comment:
Do you want to eliminate empty strings here as well or only nils? If it's reading from YAML I think it's possible you could get an empty string instead of a nil?